File: RepairDoor.cpp
#include "iostream"
#include "set"
using namespace std;

int main(){
    int n; cin>> n;
    std::multiset<int> s;
    for(int i=0; i<n; i++){
        int t; cin>> t;
        s.insert(t);
    }
    int res = 0;
    while(s.size() > 1){
        int a = *s.begin(); s.erase(s.begin());
        int b = *s.begin(); s.erase(s.begin());
        res += a+b;
        s.insert(a+b);      
    }
    cout<< res;
    return 0;
}

File: RepairDoor.py
#问题相当于带权路径长度，即哈夫曼树
n = int(input())
l = list(map(int, input().split()))
l.sort(reverse = True)
res = 0
while len(l) > 1:
    a = l.pop()
    b = l.pop()
    res = res + a+b
    l.insert(0,a+b)
    l.sort(reverse = True)
print(res)
